dbt/adapters/redshift/connections.py

```diff
@@ -323,6 +327,29 @@ def __iam_idc_browser_kwargs(credentials) -> Dict[str, Any]:
 
         return __iam_kwargs(credentials) | idc_kwargs
 
+    def __iam_idc_token_kwargs(credentials) -> Dict[str, Any]:
+        """
+        Expand the logic and flow control here to account for other Identity providers
```

Comment:
I think we can make the docstring here a little more evergreen, maybe something like: 
```suggestion
        Accepts a `credentials` object with a `token_endpoint` field that corresponds to either Okta or Entra authentication services.
```